applications/StructuralMechanicsApplication/custom_response_functions/structural_response.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import print_function, absolute_import, division
+
+# importing the Kratos Library
+from KratosMultiphysics import *
+import structural_mechanics_analysis
+
+import time as timer
+
+class ResponseFunctionBase(object):
+    def __init__(self, identifier, project_parameters):
+        self.identifier = identifier
+        self.project_parameters = project_parameters
+    def Initialize(self):
+        pass
+    def CalculateValue(self):
+        raise NotImplementedError("CalculateValue needs to be implemented by the base class")
+    def CalculateGradient(self):
+        raise NotImplementedError("CalculateGradient needs to be implemented by the base class")
+    def GetShapeGradient(self):
+        raise NotImplementedError("GetShapeGradient needs to be implemented by the base class")
+    def Finalize(self):
+        pass
+
+class SimpleResponseFunctionWrapper(ResponseFunctionBase):
```

Comment:
I dont like the terms "simple" and "wrapper" here. 

to the first term: Is it meant to be a "default" response function with default primal and adjoint runs? If yes, then I would reflect this in the name and call it "default...".

to the second term: IMO, these classes do not wrap a response function, they are actually making it up! So I would remove the term completely and just call it e.g. "MassResponseFunction". This is much more descriptive.

Reply:
The reason for the Wrapper appendix was, that our cpp classes are already called e.g. ```MassResponseFunction```.
I see your point, so i suggest to rename the cpp classes, because they are not intended to be directly used...